op-e2e/custom_gas_token_test.go

```diff
@@ -200,6 +200,127 @@ func TestCustomGasToken(t *testing.T) {
 		require.Equal(t, withdrawAmount, diff)
 	}
 
+	// checkFeeWithdrawal ensures that the FeeVault can be withdrawn from
+	checkFeeWithdrawal := func(t *testing.T, enabled bool) {
+		feeVault, err := bindings.NewSequencerFeeVault(predeploys.SequencerFeeVaultAddr, l2Client)
+		require.NoError(t, err)
+
+		// Alice will be sending transactions
+		aliceOpts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Alice, cfg.L2ChainIDBig())
+		require.NoError(t, err)
+
+		// Get the recipient of the funds
+		recipient, err := feeVault.RECIPIENT(&bind.CallOpts{})
+		require.NoError(t, err)
+
+		// This test depends on the withdrawal network being L1 which is represented
+		// by 0 in the enum.
+		withdrawalNetwork, err := feeVault.WITHDRAWALNETWORK(&bind.CallOpts{})
+		require.NoError(t, err)
+		require.Equal(t, withdrawalNetwork, uint8(0))
+
+		// Get the balance of the recipient on L1
+		var recipientBalanceBefore *big.Int
+		if enabled {
+			recipientBalanceBefore, err = weth9.BalanceOf(&bind.CallOpts{}, recipient)
+		} else {
+			recipientBalanceBefore, err = l1Client.BalanceAt(context.Background(), recipient, nil)
+		}
+		require.NoError(t, err)
+
+		// Get the min withdrawal amount for the FeeVault
+		amount, err := feeVault.MINWITHDRAWALAMOUNT(&bind.CallOpts{})
+		require.NoError(t, err)
+
+		l1opts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Alice, cfg.L1ChainIDBig())
+		require.NoError(t, err)
+		// Alice deposits funds
+		if enabled {
+			// approve + transferFrom flow
+			// Cannot use `transfer` because of the tracking of balance in the OptimismPortal
+			depositAmount := new(big.Int).Mul(amount, big.NewInt(4))
+			l1opts.Value = depositAmount
+			dep, err := weth9.Deposit(l1opts)
+			waitForTx(t, dep, err, l1Client)
+
+			l1opts.Value = nil
+			tx, err := weth9.Approve(l1opts, cfg.L1Deployments.OptimismPortalProxy, depositAmount)
+			waitForTx(t, tx, err, l1Client)
+
+			optimismPortal, err := bindings.NewOptimismPortal(cfg.L1Deployments.OptimismPortalProxy, l1Client)
+			require.NoError(t, err)
+			deposit, err := optimismPortal.DepositERC20Transaction(l1opts, cfg.Secrets.Addresses().Alice, depositAmount, depositAmount, 500_000, false, []byte{})
+			waitForTx(t, deposit, err, l1Client)
+		} else {
+			// send ether to the portal directly, alice already has funds on L2
+			optimismPortal, err := bindings.NewOptimismPortal(cfg.L1Deployments.OptimismPortalProxy, l1Client)
+			require.NoError(t, err)
+			l1opts.Value = new(big.Int).Mul(amount, big.NewInt(2))
+			tx, err := optimismPortal.DonateETH(l1opts)
+			waitForTx(t, tx, err, l1Client)
+		}
+
+		// Get Alice's balance on L2
+		aliceBalance, err := l2Client.BalanceAt(context.Background(), cfg.Secrets.Addresses().Alice, nil)
+		require.NoError(t, err)
+		require.GreaterOrEqual(t, aliceBalance.Uint64(), uint64(0))
```

Comment:
If alice is sending `amount` on [line 269](https://github.com/ethereum-optimism/optimism/pull/10685/files#diff-65937af98e6b5dabf862058a2f429f295b243fce1a1f00047e1aa3726aff9d08R269), do we not need to check that her balance is greater than `amount` here?